database/user.go

```diff
@@ -70,14 +68,21 @@ type (
 		ID              primitive.ObjectID `bson:"_id,omitempty" json:"-"`
 		Sub             string             `bson:"sub" json:"sub"`
 		Tier            int                `bson:"tier" json:"tier"`
-		StorageUsed     int64              `bson:"storage_used" json:"storageUsed"`
 		SubscribedUntil time.Time          `bson:"subscribed_until" json:"subscribedUntil"`
 	}
-	// UserDetails builds on top of User and provides additional information,
-	// typically more expensive to fetch than a single DB query.
-	UserDetails struct {
-		User
-		BandwidthUsed int64 `bson:"-" json:"bandwidthUsed"`
+	// UserStats contains statistical information about the user.
+	UserStats struct {
+		StorageUsed        int64 `bson:"-" json:"storageUsed"`
+		NumRegReads        int64 `bson:"-" json:"numRegReads"`
+		NumRegWrites       int64 `bson:"-" json:"numRegWrites"`
+		NumUploads         int   `bson:"-" json:"numUploads"`
+		NumDownloads       int   `bson:"-" json:"numDownloads"`
+		TotalUploadsSize   int64 `bson:"-" json:"totalUploadsSize"`
+		TotalDownloadsSize int64 `bson:"-" json:"totalDownloadsSize"`
+		BandwidthUploads   int64 `bson:"-" json:"bwUploads"`
```

Comment:
not a fan of the abbreviation in the fields here

Reply:
Not strictly needed here but it's still fewer bytes to send over the wire with `bandwidth` and `number` repeated over and over. I don't feel strongly about this, if you don't like it I can change to full words.